cfme/tests/infrastructure/test_quota_tagging.py

```diff
@@ -198,3 +198,61 @@ def test_quota_tagging_infra_via_services(request, appliance, provider, setup_pr
     provision_request.wait_for_request(method='ui')
     request.addfinalizer(provision_request.remove_request)
     assert provision_request.row.reason.text == "Quota Exceeded"
+
+
+@pytest.fixture(scope='module')
+def small_vm(provider, small_template_modscope):
+    vm = provider.appliance.collections.infra_vms.instantiate(random_vm_name(context='reconfig'),
+                                                              provider,
+                                                              small_template_modscope.name)
+    vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+    vm.refresh_relationships()
+    yield vm
+    vm.cleanup_on_provider()
+
+
+@pytest.fixture
+def custom_prov_data(request, prov_data, vm_name, template_name):
+    value = request.param
+    prov_data.update(value)
+    prov_data['catalog']['vm_name'] = vm_name
+    prov_data['catalog']['catalog_name'] = {'name': template_name}
+
+
+@pytest.mark.long_running
+@pytest.mark.parametrize(
+    ['custom_prov_data', 'processor_sockets', 'processor_cores_per_socket', 'total_processors',
+     'approve'],
+    [
+        [{'change': 'cores_per_socket', 'value': '4'}, '1', '4', '4', False],
+        [{'change': 'sockets', 'value': '4'}, '4', '1', '4', False],
+        [{'change': 'mem_size', 'value': '102400'}, '', '', '', True],
+    ],
+    indirect=['custom_prov_data'],
+    ids=['max_cores', 'max_sockets', 'max_memory']
+)
+def test_quota_vm_reconfigure(request, appliance, provider, setup_provider, admin_email, entities,
```

Comment:
Required:
You already inherited `provider` in `small_vm` fixture. In the test, you are not using `provider` so no need to inherit.

`entities` returning collection but not used in a test, so why are inherited that?

Reply:
'entities' fixture required to use because it helps to change 'quota_source_type' in function 'set_entity_quota_source'. This test is parameterised in 'user' and 'group' tests.